internal/pkg/siderolink/manager.go

```diff
@@ -684,6 +697,12 @@ func (manager *Manager) getLink(ctx context.Context, req *pb.ProvisionRequest, i
 
 		link := siderolink.NewLink(siderolink.Namespace, id, spec)
 
+		if token.ExtraData != nil {
```

Comment:
this nil check shouldn't be necessary

Reply:
this map can be nil if the token is plain, so I don't think we can't remove this check

Reply:
I mean we just read data from map, so nil wouldn't panic.

Reply:
I keep thinking that reading from a nil map will panic :D

Reply:
yep, tricky to remember :)